src/admin/redux/modules/email/reducer.js

```diff
@@ -81,8 +82,14 @@ const template_subs = ifElse
 
 const indexByProp = compose(indexBy, propOr(''))
 
+const greeting = member => assoc
+  ( 'greeting'
+  , member.first_name || member.title + ' ' + member.last_name
+  , member
+  )
+
 const [ primaries, secondaries ] =
-  map(indexByProp, [ 'primary_email', 'secondary_email' ])
+  map(compose(map(map(greeting)), indexByProp), [ 'primary_email', 'secondary_email' ])
```

Comment:
This line could maybe be rewritten. See if you can understand it though.

Hint: the inner map and the outer map are mapping over different functor instances. Do you know what the first one is? Do you know the second?